sklearn/callback/_computation_tree.py

```diff
@@ -0,0 +1,268 @@
+# License: BSD 3 clause
```

Comment:
@adrinjalali Discussing with @jeremiedbb IRL and while I was explaining to the sample-props PR, he was under the impression that the `MetaDataRequest` class would be similar to the `ComputationTree` in some regards. Maybe you could have a look for some inspiration :)